src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -345,17 +404,7 @@ private AsyncProtocolRequest GetOrCreateProtocolRequest(ref AsyncProtocolRequest
         //
         private void ProcessWrite(byte[] buffer, int offset, int count, LazyAsyncResult asyncResult)
         {
-            _sslState.CheckThrow(authSuccessCheck:true, shutdownCheck:true);
-            ValidateParameters(buffer, offset, count);
-
-            if (Interlocked.Exchange(ref _nestedWrite, 1) == 1)
-            {
-                throw new NotSupportedException(SR.Format(SR.net_io_invalidnestedcall, "Write", "write"));
-            }
-
-            // If this is an async operation, get the AsyncProtocolRequest to use.
-            // We do this only after we verify we're the sole write operation in flight.
-            AsyncProtocolRequest asyncRequest = GetOrCreateProtocolRequest(ref _writeProtocolRequest, asyncResult);
+            var asyncRequest = InitiateWrite(buffer, offset, count, asyncResult);
```

Comment:
Nit: per my other comments, InitiateWrite should really return void, but if for some reason it stays returning an AsyncProtocolRequest, then the ```var``` should be changed to ```AsyncProtocolRequest```.